pipenv/core.py

```diff
@@ -911,8 +911,16 @@ def do_create_virtualenv(python=None, site_packages=False, pypi_mirror=None):
         click.echo(crayons.blue("{0}".format(c.out)), err=True)
         if c.returncode != 0:
             sp.fail(environments.PIPENV_SPINNER_FAIL_TEXT.format("Failed creating virtual environment"))
+            known_exceptions = {
+                "PermissionError": "Permission denied:",
+            }
+            # PermissionError - hide the traceback for better UX
+            for partition in (part
+                              for e, part in known_exceptions.items() if e in c.err):
+                known_exceptions_partition = c.err.rpartition(partition)
+                c.err = "{} {}".format(known_exceptions_partition[1], known_exceptions_partition[2])
```

Comment:
Won't `known_exceptions_partition[1]` and `known_exceptions_partition[2]` just be `PermissionError: <message>` in  this case? I like where this is headed but I feel like it might belong in `exceptions.py`.  Also note that it really isn't appropriate to manipulate `c.err` -- it is the stored `stderr` stream output from a subprocess that has already or is currently executing.  We really should only ever manipulate how we interpret and display the result.

Possibly we should have some helper method like this:

```python
KNOWN_EXCEPTIONS = {
    "PermissionError": "Permission denied:"
}

def prettify_exc(exc):
    matched_exceptions = [k for k in KNOWN_EXCEPTIONS.keys() if k in exc]
    errors = []
    for match in matched_exceptions:
        exc, exc_name, error = exc.rpartition(KNOWN_EXCEPTIONS[match])
        errors.append("{0} {1}".format(exc_name, crayons.red(decode_for_output(error))))
    else:
        errors.append("{0}".format(decode_for_output(exc)))
    return "\n".join(errors)
```

And then the exceptions themselves can share a dictionary of known exceptions and use `prettify_exc` when `environments.is_verbose() is False`

Reply:
That's a good idea. I have pushed the ``prettify_exc`` implementation.